civis/imports/socrata_helpers.py

```diff
@@ -246,68 +277,222 @@ def Merge(dict1, dict2):
+    """
+    Selects Socrata to SQL mapping based on database that is passed in and
+    optional varchar_len argument.
+    """
     if database.lower() == "redshift":
         if varchar_len is None:
             sql_type = {
                 "number": "DOUBLE PRECISION",
-                "text": "VARCHAR(256)",
+                "double": "DOUBLE PRECISION",
+                "money": "DOUBLE PRECISION",
+                "checkbox": "boolean",
+                "text": "VARCHAR(1024)",
+                "location": "VARCHAR(1024)",
+                "multiline": "VARCHAR(1024)",
+                "multipoint": "VARCHAR(1024)",
+                "multipolygon": "VARCHAR(1024)",
+                "polygon": "VARCHAR(1024)",
                 "calendar_date": "TIMESTAMP",
-                "point": "VARCHAR(256)",
+                "point": "VARCHAR(1024)",
             }
         else:
             sql_type = {
                 "number": "DOUBLE PRECISION",
-                "text": "VARCHAR(256)",
+                "double": "DOUBLE PRECISION",
+                "money": "DOUBLE PRECISION",
+                "checkbox": "boolean",
+                "text": "VARCHAR(1024)",
+                "location": "VARCHAR(1024)",
+                "multiline": "VARCHAR(1024)",
+                "multipoint": "VARCHAR(1024)",
+                "multipolygon": "VARCHAR(1024)",
+                "polygon": "VARCHAR(1024)",
                 "calendar_date": "TIMESTAMP",
-                "point": "VARCHAR(256)",
+                "point": "VARCHAR(1024)",
             }
 
             varchar_len = "VARCHAR(" + varchar_len + ")"
 
             sql_type["point"] = varchar_len
             sql_type["text"] = varchar_len
+            sql_type["location"] = varchar_len
 
     elif database.lower() == "postgres":
         if varchar_len is None:
             sql_type = {
                 "number": "DOUBLE PRECISION",
-                "text": "VARCHAR(256)",
+                "double": "DOUBLE PRECISION",
+                "money": "DOUBLE PRECISION",
+                "checkbox": "boolean",
+                "text": "VARCHAR(1024)",
+                "location": "VARCHAR(1024)",
+                "multiline": "VARCHAR(1024)",
+                "multipoint": "VARCHAR(1024)",
+                "multipolygon": "VARCHAR(1024)",
+                "polygon": "VARCHAR(1024)",
                 "calendar_date": "TIMESTAMP",
                 "point": "POINT",
             }
         else:
             sql_type = {
                 "number": "DOUBLE PRECISION",
-                "text": "VARCHAR(256)",
+                "double": "DOUBLE PRECISION",
+                "money": "DOUBLE PRECISION",
+                "checkbox": "boolean",
+                "text": "VARCHAR(1024)",
+                "location": "VARCHAR(1024)",
+                "multiline": "VARCHAR(1024)",
+                "multipoint": "VARCHAR(1024)",
+                "multipolygon": "VARCHAR(1024)",
+                "polygon": "VARCHAR(1024)",
```

Comment:
I hadn't come cross any of these datatypes in the two datasets that I worked with. From the API docs, it seems as if the way Socrata outputs these geometries are not readily readable by PostGres (just like POINT) and will require some gymnastics. Since I had no sample data to work with I wanted to ensure reliability for now and work on these geometries as they come up. And longer term - as engineering takes over this import these transformations will be included in the roadmap!